CMakeLists.txt

```diff
@@ -797,7 +801,7 @@ set(zephyr_lnk
   ${ZEPHYR_LIBS_PROPERTY}
   ${LINKERFLAGPREFIX},--no-whole-archive
   kernel
-  ${OFFSETS_O_PATH}
+  $<TARGET_OBJECTS:offsets>                                                 #                zephyr/CMakeFiles/offsets.dir/arch/xmcu/core/offsets/offsets.c.obj
```

Comment:
It's good to see that we can get rid of

 ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/offsets.dir/arch/${ARCH}/core/offsets/offsets.c.obj

I can approve this if you remove this comment and move some of the PR description text into the commit message text.